Extensions/BitBucket/Src/Tasks/DownloadArtifactsBitbucket/downloadBitbucket.js

```diff
@@ -97,23 +114,36 @@ https.request(options, function (rs) {
 
 function getEndpointDetails(inputFieldName) {
     var bitbucketEndpoint = tl.getInput(inputFieldName);
-    var hostUsername = getAuthParameter(bitbucketEndpoint, 'username');
-    var hostPassword = getAuthParameter(bitbucketEndpoint, 'password');
-    tl.debug('hostUsername: ' + hostUsername);
-    tl.debug('hostPassword: ' + hostPassword);
-
-    return {
-        "Username": hostUsername,
-        "Password": hostPassword
-    };
+    var auth = tl.getEndpointAuthorization(bitbucketEndpoint, false);
+    var scheme = auth.scheme.toLowerCase().trim();
+    
+    if (scheme === "token") {
+        var token = getAuthParameter(bitbucketEndpoint, 'apitoken');
+        var username = getAuthParameter(bitbucketEndpoint, 'email') || '';
+        tl.debug('Using token authentication');
+        return {
+            "Token": token,
+            "Username": username
+        };
+    } else {
+        var hostUsername = getAuthParameter(bitbucketEndpoint, 'username');
+        var hostPassword = getAuthParameter(bitbucketEndpoint, 'password');
+        tl.debug('hostUsername: ' + hostUsername);
+        tl.debug('hostPassword: ' + hostPassword);
+        return {
+            "Username": hostUsername,
+            "Password": hostPassword
+        };
+    }
 }
 
 function getAuthParameter(endpoint, paramName) {
     var paramValue = null;
     var auth = tl.getEndpointAuthorization(endpoint, false);
```

Comment:
`auth `can be passed as parameter here, since there is no need to resolve it multiple times. Since this is not something you introduced take this as optional.